webapp/hooks/useI18n.ts

```diff
@@ -1,5 +1,15 @@
+import i18nInstance from '@common/i18n/i18nInstance'
 import { i18n } from 'i18next'
-
+import { useEffect, useState } from 'react'
 import { useAppSelector } from '@webapp/store'
 
-export default (): i18n => useAppSelector((state) => state.app.i18n)
+export default (): Partial<i18n> => {
+  const [instance, setInstance] = useState(i18nInstance.getInstance())
```

Comment:
why adding to local components state the i18n instance ?
have you checked react i18 next? https://react.i18next.com